cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/hdfs/HdfsStorageClient.java

```diff
@@ -0,0 +1,59 @@
+package com.linkedin.openhouse.cluster.storage.hdfs;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.openhouse.cluster.storage.StorageClient;
+import com.linkedin.openhouse.cluster.storage.StorageType;
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import java.io.IOException;
+import javax.annotation.PostConstruct;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.fs.FileSystem;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+/**
+ * The HdfsStorageClient class is an implementation of the StorageClient interface for HDFS storage.
+ * It uses an Apache Hadoop FileSystem to interact with the HDFS file system.
+ */
+@Slf4j
+@Lazy
+@Component
+public class HdfsStorageClient implements StorageClient<FileSystem> {
+
+  private FileSystem fs;
+
+  @Autowired private StorageProperties storageProperties;
+
+  private static final StorageType.Type HDFS_TYPE = StorageType.HDFS;
+
+  /** Initialize the HdfsStorageClient when the bean is accessed for the first time. */
+  @PostConstruct
+  public synchronized void init() throws IOException {
+    log.info("Initializing storage client for type: " + HDFS_TYPE);
+    Preconditions.checkArgument(
+        !(storageProperties.getTypes() == null
+            || storageProperties.getTypes().isEmpty()
+            || !storageProperties.getTypes().containsKey(HDFS_TYPE.getValue())),
+        "Storage properties doesn't contain type: " + HDFS_TYPE.getValue());
+    StorageProperties.StorageTypeProperties hdfsStorageProperty =
+        storageProperties.getTypes().get(HDFS_TYPE.getValue());
+    Preconditions.checkArgument(
+        hdfsStorageProperty != null,
+        "Storage properties doesn't contain type: " + HDFS_TYPE.getValue());
+    Preconditions.checkArgument(
+        hdfsStorageProperty.getEndpoint() != null,
+        "Storage properties doesn't contain endpoint for: " + HDFS_TYPE.getValue());
+    Preconditions.checkArgument(
+        hdfsStorageProperty.getRootPath() != null,
+        "Storage properties doesn't contain rootpath for: " + HDFS_TYPE.getValue());
+    org.apache.hadoop.conf.Configuration configuration = new org.apache.hadoop.conf.Configuration();
+    configuration.set("fs.defaultFS", hdfsStorageProperty.getEndpoint());
+    fs = FileSystem.get(configuration);
```

Comment:
@abhisheknath2011 how is our hdfs conf loaded today?

Reply:
We bake in Hadoop/Hdfs conf as part of docker image and that gets mounted to the services pod as part of deployment. This should work provided we inject hadoop dependencies and set appropriate hadoop classpath during deployment.